src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryServiceSupport.scala

```diff
@@ -69,11 +69,11 @@ trait LibraryServiceSupport extends DataUseRestrictionSupport with LazyLogging {
 
       val annotatedWorkspaces = workspaces map { ws =>
         // does this workspace have an orsp id?
-        ws.attributes.get(orspIdAttribute) match {
+        ws.attributes.getOrElse(Map.empty).get(orspIdAttribute) match {
           case Some(s:AttributeString) =>
             val orspAttrs = restrictionMap.getOrElse(s.value, Map.empty[AttributeName, Attribute])
-            val newAttrs = replaceDataUseAttributes(ws.attributes, orspAttrs)
-            ws.copy(attributes = newAttrs)
+            val newAttrs = replaceDataUseAttributes(ws.attributes.getOrElse(Map.empty), orspAttrs)
+            ws.copy(attributes = Option(newAttrs))
```

Comment:
`Option(newAttrs))` or `Some(newAttrs))` ?

Reply:
The way you originally had it is generally preferred because it safely handles null values. `Option(null)` will equate to `None`, but `Some(null)` will throw an exception. I pretty much always use `Option(foo)` now because of this, even if I know that `foo` isn't going to be null

Reply:
good to know -- thanks for the info!